core/__tests__/tasks/system/destroy.ts

```diff
@@ -226,7 +226,7 @@ describe("tasks/destroy", () => {
                 type: "integer",
                 mappedColumn: "id",
               });
-              await userIdProperty.update({ isPrimaryKey: true });
+              await source.setMapping({ id: "userId" });
```

Comment:
@rwfeather This was why those tests started failing - the `.update()` call triggered some other hooks that tried to get the mapping, but the old mapped property no longer existed so errors were being thrown and `isPrimaryKey` was never being set. Properly setting the mapping on the source to the newly created property and letting `isPrimaryKey` be determined as normal lines everything up so the mapping on the source itself isn't referring to something that no longer exists.